R/vars_manager.R

```diff
@@ -139,91 +158,108 @@ get_matrix_var <- function(
+                        }
+                        if (is.numeric(max) && col_numeric) {
+                            obj_filtered <- obj_filtered[obj_filtered[,col_name] <= max,]
+                        }
                     }
                 }
             }
         }
-    }
 
-    abbrev <- !is.null(abbrev_len) && length(obj_filtered) > abbrev_len
-    dims <- dim(obj_filtered)
+        dims <- dim(obj_filtered)
 
-    if (is.vector(sort_by) && length(sort_by) > 0) {
-        cols <- obj_filtered[,sort_by]
-        order_params <- as.list(as.data.frame(cols))
+        start <- `if`(!is.null(page_size), page_size * page + 1, 1)
+        has_next_page <- `if`(!is.null(page_size), dims[[1]] > (start + page_size - 1), FALSE)
+        end <- `if`(has_next_page, start + page_size - 1, dims[[1]])
 
-        if (is.vector(ascending) && length(ascending) == length(sort_by)) {
-            i <- 0
-            for (col in order_params) {
-                i<-i+1
-                order_params[[i]] <- `if`(ascending[[i]], order_params[[i]], order_params[[i]]*-1)
-            }
-        } else if (is.logical(ascending) && ascending == FALSE) {
-            i <- 0
-            for (col in order_params) {
-                i <- i+1
-                order_params[[i]] <- order_params[[i]]*-1
+        if (is.vector(sort_by) && length(sort_by) > 0) {
+            cols <- obj_filtered[,sort_by]
+            order_params <- as.list(as.data.frame(cols))
+
+            if (is.vector(ascending) && length(ascending) == length(sort_by)) {
+                i <- 0
+                for (col in order_params) {
+                    i<-i+1
+                    order_params[[i]] <- `if`(ascending[[i]], order_params[[i]], order_params[[i]]*-1)
+                }
+            } else if (is.logical(ascending) && ascending == FALSE) {
+                i <- 0
+                for (col in order_params) {
+                    i <- i+1
+                    order_params[[i]] <- order_params[[i]]*-1
+                }
             }
+
+            obj_sorted <- obj_filtered[do.call(order, order_params),]
+            obj_pre <- obj_sorted[start:end,]
+        } else {
+            obj_pre <- obj_filtered[start:end,]
         }
 
-        obj_sorted <- obj_filtered[do.call(order, order_params),]
-        obj_pre <- `if`(abbrev, obj_sorted[1:abbrev_len,], obj_sorted)
-    } else {
-        obj_pre <- `if`(abbrev, obj_filtered[1:abbrev_len,], obj_filtered)
-    }
+        data <- list()
+        for (i in 1:nrow(obj_pre)) {
+            data[[i]] = as.list(as.character(obj_pre[i,]))
+        }
+        col_names <- list()
+        i <- 1
+        for (col_name in 1:ncol(obj_pre)) {
+            col_names[i] <- as.character(col_name)
+            i <- i + 1
+        }
+        row_names <- list()
+        i <- 1
+        for (row_name in 1:nrow(obj_pre)) {
+            row_names[i] <- as.character(row_name)
+            i <- i + 1
```

Comment:
can this be combined with the for loop on line 233?

Reply:
same concern as above, this doesn't account for the page offset, the row_names should match the page you're on